apps/ios/Shared/Views/Chat/ComposeMessage/ComposeView.swift

```diff
@@ -232,43 +245,47 @@ struct ComposeView: View {
         VStack(spacing: 0) {
             contextItemView()
             switch (composeState.editing, composeState.preview) {
-                case (true, .filePreview): EmptyView()
-                case (true, .voicePreview): EmptyView() // ? we may allow playback when editing is allowed
-                default: previewView()
+            case (true, .filePreview): EmptyView()
+            case (true, .voicePreview): EmptyView() // ? we may allow playback when editing is allowed
+            default: previewView()
             }
             HStack (alignment: .bottom) {
                 Button {
                     showChooseSource = true
                 } label: {
                     Image(systemName: "paperclip")
-                        .resizable()
+                            .resizable()
                 }
-                .disabled(composeState.attachmentDisabled)
-                .frame(width: 25, height: 25)
-                .padding(.bottom, 12)
-                .padding(.leading, 12)
+                        .disabled(composeState.attachmentDisabled)
+                        .frame(width: 25, height: 25)
+                        .padding(.bottom, 12)
+                        .padding(.leading, 12)
                 SendMessageView(
-                    composeState: $composeState,
-                    sendMessage: {
-                        sendMessage()
-                        resetLinkPreview()
-                    },
-                    sendLiveMessage: sendLiveMessage,
-                    updateLiveMessage: updateLiveMessage,
-                    voiceMessageAllowed: chat.chatInfo.featureEnabled(.voice),
-                    showEnableVoiceMessagesAlert: chat.chatInfo.showEnableVoiceMessagesAlert,
-                    startVoiceMessageRecording: {
-                        Task {
-                            await startVoiceMessageRecording()
-                        }
-                    },
-                    finishVoiceMessageRecording: finishVoiceMessageRecording,
-                    allowVoiceMessagesToContact: allowVoiceMessagesToContact,
-                    onImagesAdded: { images in if !images.isEmpty { chosenImages = images }},
-                    keyboardVisible: $keyboardVisible
+                        composeState: $composeState,
+                        sendMessage: {
+                            sendMessage()
+                            resetLinkPreview()
+                        },
+                        sendLiveMessage: sendLiveMessage,
+                        updateLiveMessage: updateLiveMessage,
+                        cancelLiveMessage: {
+                            composeState = composeState.copy(liveMessage: nil)
+                            chatModel.removeLiveChatItemDummy()
+                        },
```

Comment:
there is no need to pass this function as a parameter given that composeState is passed to SendMessageView and that the model is available there too

Reply:
you can always include it via view definition or via ChatModel.shared - it's a global singleton

Reply:
In this case it's also EnvironmentObject - so cleaner to get it there.